soc/arm64/nxp_layerscape/ls1046a/mmu_regions.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <devicetree.h>
+#include <soc.h>
+#include <sys/util.h>
+#include <arch/arm64/arm_mmu.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 0),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 1),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("DRAM",
+			      CONFIG_SRAM_BASE_ADDRESS,
+			      CONFIG_SRAM_SIZE * 1024,
+			      MT_NORMAL | MT_RW | MT_NS),
```

Comment:
Under normal circumstances there would be no need to map the whole SRAM. Any reason why you are mapping the whole space?

Reply:
I only map the SRAM used by Zephyr (1M bytes), any other concerns? thanks.

Jiafei.

Reply:
In general there would be no reason to map any SRAM region at all since all the relevant regions are already mapped when MMU is initialized.

Reply:
I found the following MMU mapping table for ARM64, so these three regisons will cover the whole SRAM space defined by CONFIG_SRAM_SIZE and CONFIG_SRAM_BASE_ADDRESS? thanks.

 620 static const struct arm_mmu_flat_range mmu_zephyr_ranges[] = {
 621
 622         /* Mark the zephyr execution regions (data, bss, noinit, etc.)
 623          * cacheable, read-write
 624          * Note: read-write region is marked execute-never internally
 625          */
 626         { .name  = "zephyr_data",
 627           .start = _image_ram_start,
 628           .end   = _image_ram_end,
 629           .attrs = MT_NORMAL | MT_P_RW_U_NA | MT_DEFAULT_SECURE_STATE },
 630
 631         /* Mark text segment cacheable,read only and executable */
 632         { .name  = "zephyr_code",
 633           .start = _image_text_start,
 634           .end   = _image_text_end,
 635           .attrs = MT_NORMAL | MT_P_RX_U_RX | MT_DEFAULT_SECURE_STATE },
 636
 637         /* Mark rodata segment cacheable, read only and execute-never */
 638         { .name  = "zephyr_rodata",
 639           .start = _image_rodata_start,
 640           .end   = _image_rodata_end,
 641           .attrs = MT_NORMAL | MT_P_RO_U_RO | MT_DEFAULT_SECURE_STATE },
 642 };

Reply:
No, the three regions are the minimum set necessary to have Zephyr running and these will not cover the whole SRAM space. The remaining SRAM space is mapped at run-time only when needed using usually `arch_mem_map()` or `arch_mem_domain_partition_add()` but that's done by the kernel and not something the user should be concerned about.

The point is that there should be very few reasons for you to map statically the whole SRAM at boot since the code is smart enough to dynamically map the SRAM regions only when needed. It is indeed so rare that someone wants to map the whole SRAM at boot that a special kconfig for that exists, see `CONFIG_ARCH_MAPS_ALL_RAM`.

Reply:
I see, will remove sram mapping, thanks.